commons-numbers-examples/examples-jmh/src/main/java/org/apache/commons/numbers/examples/jmh/core/EuclideanNormAlgorithms.java

```diff
@@ -0,0 +1,751 @@
+                    final double s = s2 + p;
+                    final double cs = DoublePrecision.twoSumLow(s2, p, s);
+
+                    // update running totals
+                    c2 += cp + cs;
+                    s2 = s;
+                }
+            }
+
+            if (s1 != 0) {
+                // add s1, s2, c1, c2
+                s2 = s2 * SCALE_DOWN * SCALE_DOWN;
+                c2 = c2 * SCALE_DOWN * SCALE_DOWN;
+                final double sum = s1 + s2;
+                c1 += DoublePrecision.twoSumLow(s1, s2, sum) + c2;
+                return Math.sqrt(sum + c1) * SCALE_UP;
+            } else if (s2 != 0) {
+                // add s2, s3, c2, c3
+                s3 = s3 * SCALE_DOWN * SCALE_DOWN;
+                c3 = c3 * SCALE_DOWN * SCALE_DOWN;
+                final double sum = s2 + s3;
+                c2 += DoublePrecision.twoSumLow(s2, s3, sum) + c3;
+                return Math.sqrt(sum + c2);
+            }
+            // add s3, c3
+            return Math.sqrt(s3 + c3) * SCALE_DOWN;
+        }
+
+        /**
+         * Compute the low part of the double length number {@code (z,zz)} for the exact
+         * square of {@code x} using Dekker's mult12 algorithm. The standard
+         * precision product {@code x*x} must be provided. The number {@code x}
+         * is split into high and low parts using Dekker's algorithm.
+         *
+         * <p>Warning: This method does not perform scaling in Dekker's split and large
+         * finite numbers can create NaN results.
+         *
+         * @param x The factor.
+         * @param xx Square of the factor (x * x).
+         * @return the low part of the product double length number
+         */
+        private static double productLowUnscaled(double x, double xx) {
+            // Split the numbers using Dekker's algorithm without scaling
+            final double hx = DoublePrecision.highPartUnscaled(x);
+            final double lx = x - hx;
+
+            return DoublePrecision.productLow(hx, lx, hx, lx, xx);
+        }
+    }
+
+    /** Modification of {@link ExtendedPrecisionLinearCombination} that only uses a single pass through
+     * the input array as well as an extended precision sqrt computation.
+     */
+    static final class ExtendedPrecisionLinearCombinationSqrt2 implements ToDoubleFunction<double[]> {
+
+        /** Threshold for scaling small numbers. */
+        private static final double SMALL_THRESH = 0x1.0p-500;
+
+        /** Threshold for scaling large numbers. */
+        private static final double LARGE_THRESH = 0x1.0p+500;
+
+        /** Value used to scale down large numbers. */
+        private static final double SCALE_DOWN = 0x1.0p-600;
+
+        /** Value used to scale up small numbers. */
+        private static final double SCALE_UP = 0x1.0p+600;
+
+        /** {@inheritDoc} */
+        @Override
+        public double applyAsDouble(final double[] v) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method contains 51 lines of code, not including blank lines or lines with only comments, Java punctuation characters, identifiers, or literals. By comparison, 98% of the methods in the CodeGuru reference dataset contain fewer lines of code. Large methods might be difficult to read and have logic that is hard to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 633-701 into a separate method.